### ISSUE TITLE
Building error with xor-mp 

### ISSUE THREAD

**Issue by duyvuleo** [clab/dynet#976]
Hi all,

Today (12 Oct), I just git-pulled the latest version of dynet, built and got the following errors:

```
CMakeFiles/xor-simple-mp.dir/multiprocessing/train_xor-simple-mp.cc.o: In function `boost::date_time::base_time<boost::posix_time::ptime, boost::date_time::counted_time_system<boost::date_time::counted_time_rep<boost::posix_time::millisec_posix_time_system_config> > >::base_time(boost::gregorian::date const&, boost::posix_time::time_duration const&, boost::date_time::dst_flags)':
/usr/include/boost/date_time/time.hpp:64: undefined reference to `dynet::mp::stop_requested'
CMakeFiles/xor-simple-mp.dir/multiprocessing/train_xor-simple-mp.cc.o: In function `SufficientStats dynet::mp::read_data<SufficientStats>(int)':
/home/vhoang2/works/dynet-clab/dynet-v2/dynet/dynet/mp.h:90: undefined reference to `dynet::mp::stop_requested'
CMakeFiles/xor-simple-mp.dir/multiprocessing/train_xor-simple-mp.cc.o: In function `int dynet::mp::run_child<Datum, SufficientStats>(unsigned int, dynet::mp::ILearner<Datum, SufficientStats>*, dynet::Trainer*, std::vector<dynet::mp::Workload, std::allocator<dynet::mp::Workload> >&, std::vector<Datum, std::allocator<Datum> > const&, std::vector<Datum, std::allocator<Datum> > const&)':
/home/vhoang2/works/dynet-clab/dynet-v2/dynet/dynet/mp.h:217: undefined reference to `dynet::mp::queue_name[abi:cxx11]'
/home/vhoang2/works/dynet-clab/dynet-v2/dynet/dynet/mp.h:254: undefined reference to `dynet::mp::shared_object'
CMakeFiles/xor-simple-mp.dir/multiprocessing/train_xor-simple-mp.cc.o: In function `boost::interprocess::mapped_region::mapped_region()':
/usr/include/boost/interprocess/mapped_region.hpp:577: undefined reference to `dynet::mp::queue_name[abi:cxx11]'
CMakeFiles/xor-simple-mp.dir/multiprocessing/train_xor-simple-mp.cc.o: In function `SufficientStats dynet::mp::run_mp_minibatch<Datum, SufficientStats>(unsigned int, dynet::mp::ILearner<Datum, SufficientStats>*, std::vector<Datum, std::allocator<Datum> > const&)':
/home/vhoang2/works/dynet-clab/dynet-v2/dynet/dynet/mp.h:407: undefined reference to `dynet::mp::generate_queue_name[abi:cxx11]()'
CMakeFiles/xor-simple-mp.dir/multiprocessing/train_xor-simple-mp.cc.o: In function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::operator=(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)':
/usr/include/c++/5/bits/basic_string.h:589: undefined reference to `dynet::mp::queue_name[abi:cxx11]'
```

Is there any changes on it? The previous ones worked fine for me. 

I am looking for your advice. Thanks!

**Comment by duyvuleo** [comment]
refers to this #973 as well.

**Comment by neubig** [comment]
Fixed by #978
